agent/http.go

```diff
@@ -161,28 +160,55 @@ func (fs *redirectFS) Open(name string) (http.File, error) {
 	return file, err
 }
 
-type templatedIndexFS struct {
-	fs           http.FileSystem
-	templateVars func() map[string]interface{}
+type settingsInjectedIndexFS struct {
+	fs         http.FileSystem
+	UISettings map[string]interface{}
 }
 
-func (fs *templatedIndexFS) Open(name string) (http.File, error) {
+func (fs *settingsInjectedIndexFS) Open(name string) (http.File, error) {
 	file, err := fs.fs.Open(name)
 	if err != nil || name != "/index.html" {
 		return file, err
 	}
 
-	content, _ := ioutil.ReadAll(file)
+	content, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed reading index.html: %s", err)
+	}
 	file.Seek(0, 0)
-	t, err := template.New("fmtedindex").Parse(string(content))
+
+	// Replace the placeholder in the meta ENV with the actual UI config settings.
+	// Ember passes the ENV with URL encoded JSON in a meta tag. We are replacing
+	// a value that is the encoded version of "__CONSUL_UI_SETTINGS_GO_HERE__"
```

Comment:
Need to update this comment.

```suggestion
	// a key and value that is the encoded version of `"CONSUL_UI_SETTINGS":"__CONSUL_UI_SETTINGS_GO_HERE__"`
```